packages/dev/core/src/Engines/engine.ts

```diff
@@ -421,7 +421,22 @@ export class Engine extends ThinEngine {
     /**
      * Gets a boolean indicating if the pointer is currently locked
      */
-    public isPointerLock = false;
+    public get isPointerLock() {
+        let canvasCheck = false;
+        // Check if document exists and update canvasCheck if it does
+        if (IsDocumentAvailable() && this._renderingCanvas) {

```

Comment:
Since the pointerlock can be programmatically initiated or ended at any point during an input and a `pointerlockchange` event will occur after pointer input events, this means that there's a chance that at least one event can be erroneously processed as a pointer locked event.  This was the option that came to mind to cover all scenarios.

Reply:
(IsDocumentAvailable() && this._renderingCanvas) should never change after initialization I guess so maybe you could store document or a shim of document in the constructor so that the full test and compare becomes 

return this._isPointerLock || this._documentShim.pointerLockElement === this._renderingCanvas;

Reply:
Maybe good to check if the condition works in Babylon Native also.

Reply:
Small nit - there is no need to check if `renderingCanvas` is available - the next line will be false if it is undefined.

Reply:
@bghgary  AFAIK, document isn't defined in Babylon Native but it is in the playground (just adds a basic empty object); however, renderingCanvas should always be undefined.  That's why I added both of those checks.  If both happen to be undefined, `document.pointerLockElement === this._renderingCanvas` will resolve to true (caught during test running).  If we do get pointerlock implemented for Native, we can always update this if statement or just manually set it (overrides value).

@sebavan I can clean up the bools to just verify the existence of a document and canvas file once.  I just need to account for Native possibly checking `undefined === undefined`

Reply:
I've updated the logic and added a test.